federatedscope/cl/trainer/trainer.py

```diff
@@ -0,0 +1,111 @@
+from federatedscope.core.auxiliaries.enums import MODE
+from federatedscope.register import register_trainer
+from federatedscope.core.auxiliaries.optimizer_builder import get_optimizer
+from federatedscope.core.auxiliaries.scheduler_builder import get_scheduler
+from federatedscope.core.trainers import GeneralTorchTrainer
+from federatedscope.core.trainers.context import CtxVar
+from federatedscope.core.auxiliaries.enums import LIFECYCLE
+from federatedscope.core.auxiliaries import utils
+import torch
+import numpy as np
+
+
+class CLTrainer(GeneralTorchTrainer):
+    def __init__(self,
+                 model,
+                 data,
+                 device,
+                 config,
+                 only_for_eval=False,
+                 monitor=None):
+        super(CLTrainer, self).__init__(model, data, device, config,
+                                              only_for_eval, monitor)
+        self.batches_aug_data_1, self.batches_aug_data_2 = [], []
+
+    @torch.no_grad()
+    def get_train_pred_embedding(self):
+        model = self.ctx.model.to(self.ctx.device)
```

Comment:
here, make sure the model is in the `train` mode

Reply:
this function `get_train_pred_embedding` is in a different style w.r.t. those basic ones of a trainer, e.g., `train`, `eval`, etc. em... it is ok to me, but have you ensured it is in `train` mode?